### PR TITLE
Document bar

### PR DESCRIPTION
**Pull request by brendankmorrison**
check comments

### CODE REVIEW

**Comment by nneven** [comment]
> 18 files changed

> 2 commits

> based

File: components/Documentbar/Documentbar.context.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { DocumentbarInitialState } from './Documentbar.state';
 export interface DocumentbarContextProps {
   state: DocumentbarInitialState;
   dispatch: Dispatch<ActionType<DocumentbarInitialState>>;
-  handleCreateDocument: () => void;
+  handleCreateDocument: (event: ChangeEvent<HTMLInputElement>) => void;
```
Comment (nneven) [inline components/Documentbar/Documentbar.context.tsx:12]: do you understand what is a `ChangeEvent`? My first time seeing so I might need to read up on it

File: components/DocumentViewer/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default } from './DocumentViewer';
```
Comment (nneven) [inline components/DocumentViewer/index.ts:1]: is this necessary? other components such as `Chatbar` do not have an `index.ts`, I wonder what the standard practice is

File: components/Documentbar/components/DocumentFolders.tsx [hunk 3]
```diff
@@ -39,7 +39,9 @@ export const DocumentFolders = () => {
         if (document.folderId === currentFolder.id) {
           return (
             <div key={index} className="ml-5 gap-2 border-l pl-2">
+              <div>Homework</div>
```
Comment (nneven) [inline components/Documentbar/components/DocumentFolders.tsx]: delete maybe? idk what this is for

File: package.json [hunk 4]
```diff
@@ -15,6 +15,8 @@
     "@dqbd/tiktoken": "^1.0.2",
     "@tabler/icons-react": "^2.9.0",
     "eventsource-parser": "^0.1.0",
+    "formidable": "^2.1.1",
+    "fs": "^0.0.1-security",
```
Comment (nneven) [inline package.json:19]: what is formidable and fs and where did they come from? curious why you would need new npm packages

**Review by nneven — CHANGES_REQUESTED** [review]
Looks great overall! I left a few questions regarding small things which we can address in-person or we can continue the discussion here before merging.

File: components/Documentbar/Documentbar.context.tsx (hunk 1 above)
Comment (brendankmorrison) [reply components/Documentbar/Documentbar.context.tsx:12]: the event is a file upload from a form

File: components/DocumentViewer/index.ts (hunk 2 above)
Comment (brendankmorrison) [reply components/DocumentViewer/index.ts:1]: not sure

File: components/Documentbar/components/DocumentFolders.tsx (hunk 3 above)
Comment (brendankmorrison) [reply components/Documentbar/components/DocumentFolders.tsx]: yes, good catch

File: package.json (hunk 4 above)
Comment (brendankmorrison) [reply package.json:19]: file stuff sending files as form in post request

File: components/DocumentViewer/index.ts (hunk 2 above)
Comment (nneven) [reply components/DocumentViewer/index.ts:1]: its ok, we can keep for now and easily remove later, I don't think its necessary fwiw but not sure why some of the components have it and some don't

**Review by nneven — APPROVED** [review]

**Pull request merged**